include/zephyr/rtio/rtio.h

```diff
@@ -999,8 +987,8 @@ static inline uint32_t rtio_cqe_compute_flags(struct rtio_iodev_sqe *iodev_sqe)
 #ifdef CONFIG_RTIO_SYS_MEM_BLOCKS
 	if (iodev_sqe->sqe.op == RTIO_OP_RX && iodev_sqe->sqe.flags & RTIO_SQE_MEMPOOL_BUFFER) {
 		struct rtio *r = iodev_sqe->r;
-		struct sys_mem_blocks *mem_pool = r->block_pool->mempool;
-		uint32_t block_size = r->block_pool->blk_size;
+		struct sys_mem_blocks *mem_pool = r->block_pool;
+		uint32_t block_size = UINT32_C(1) << mem_pool->blk_sz_shift;
```

Comment:
Nit: we calculate `block_size` from `blk_sz_shift` in a few places, can we create a macro for this?

Reply:
in this case a static inline should suffice, no need for a macro

Reply:
Done